contrib/check-image-size.py

```diff
@@ -0,0 +1,147 @@
+    for profile_name, profile_data in profile_json["profiles"].items():
+        device_profile = {
+            "name": profile_name,
+        }
+        if "image_size" in profile_data.get("file_size_limits", {}):
+            device_profile["max_image_size"] = profile_data["file_size_limits"]["image"]
+
+        for image in profile_data["images"]:
+            if image["type"] != "sysupgrade":
+                continue
+            if "size" in image:
+                device_profile["image_size"] = image["size"]
+        
+        profiles.append(device_profile)
+    
+    return profiles
+
+
+def check_image_size_below_limit(profile, overhead=0):
+    # Skip devices that pad their images
+    if profile["name"] in excluded_devices:
+        return ImageSizeCheck.IGNORED
+
+    if "max_image_size" in profile and "image_size" in profile:
+        if profile["image_size"] + (overhead * 1024) > profile["max_image_size"]:
+            return ImageSizeCheck.TOO_BIG
+        else:	
+            return ImageSizeCheck.OK
+    
+    return ImageSizeCheck.UNKNOWN
+
+
+def print_github_actions_warning(message):
+    print('::warning::{}'.format(message))
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Check image size of OpenWrt profiles')
+    parser.add_argument(
+        'profile_json',
+        help='Path to profile.json',
+        nargs='+'
+    )
+    parser.add_argument(
+        '--github-actions',
+        help='Generate warnings for use with GitHub Actions',
+        action='store_true'
+    )
+    parser.add_argument(
+        '--overhead',
+        type=int,
+        help='Additional size to add to the image size in kilobyte',
+        default=0
+    )
+    args = parser.parse_args()
+
+    if args.profile_json is None:
+        print('Error: profile.json not specified')
+        sys.exit(1)
+    
+    # Load all profile.json files
+    profiles = []
+    for profile_file in args.profile_json:
+        profiles.extend(load_openwrt_profile_json(profile_file))
+
+    # Initialize results with all available ImageSizeCheck values
+    results = {}
+    for check_result in ImageSizeCheck:
+        results[check_result] = []
+
+    for profile in profiles:
+        check_result = check_image_size_below_limit(profile, args.overhead)
+        results[check_result].append(profile)
+    
+    for check_result, profiles in results.items():
+        if len(profiles) == 0:
+            continue
+
+        # Group by result type for GitHub Actions
+        if args.github_actions:
+            print('::group::{}'.format(check_result.value))
+
+        for profile in profiles:
+            if check_result == ImageSizeCheck.TOO_BIG:
+                msg = 'Image size of profile {} is too big ({} > {})'.format(
+                    profile["name"],
+                    profile["image_size"] + (args.overhead * 1024),
+                    profile["max_image_size"])
+                print_warning(args.github_actions, msg)
```

Comment:
`print_warning` is not defined

Reply:
it's probably an artifact from the previous solution and was replaced by line 130 and 131
So it's enough to remove line 129.